### ISSUE TITLE
localhost errors

### ISSUE THREAD

**Issue by walidmujahid** [gitcoinco/web#7347]
My local build seems to be erroring out. When trying to load in the borwser I recieve a `This page isn’t working localhost didn’t send any data. ERR_EMPTY_RESPONSE`

After looking through the logs, I think, but am uncertain, that it might be the new elasticsearch feature - possible bug introduced in #7204, #7307, or another.

Logs:

```
elasticsearch_1  | level=info ts=2020-09-10T02:37:00.303391064Z caller=clusterinfo.go:166 msg="providing consumers with updated cluster info label"
elasticsearch_1  | level=error ts=2020-09-10T02:37:01.10406585Z caller=clusterinfo.go:250 msg="failed to get cluster info" err="Get http://localhost:9200/: dial tcp 127.0.0.1:9200: connect: connection refused"
elasticsearch_1  | level=error ts=2020-09-10T02:37:01.326867946Z caller=clusterinfo.go:171 msg="failed to retrieve cluster info from ES" err="Get http://localhost:9200/: dial tcp 127.0.0.1:9200: connect: connection refused"
elasticsearch_1  | [2020-09-10T02:37:45,826][INFO ][o.e.n.Node               ] [] initializing ...
elasticsearch_1  | level=info ts=2020-09-10T02:42:01.646918002Z caller=clusterinfo.go:166 msg="providing consumers with updated cluster info label"
elasticsearch_1  | level=error ts=2020-09-10T02:42:09.526143221Z caller=clusterinfo.go:250 msg="failed to get cluster info" err="Get http://localhost:9200/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
```

I ran grep for anything related to elasticsearch, I came accross:

https://github.com/gitcoinco/web/blob/4ecca77fa6694e7f37b11a295f0997d215d5002e/app/app/settings.py#L831

Currently the image used is elasticsearch5. I do not know the implementation details, however, I quickly tested repalcing the 2 with a 5 for L831 in settings, I was unable to correct the errors by doing this. However, could this line in Settings coupled with something else be causing these local errors?

**Comment by walidmujahid** [comment]
@androolloyd

**Comment by owocki** [comment]
@walidmujahid are there any bugs coming from the web_web container?  you should be able to run the web container with out connecting to elasticserach

**Comment by androolloyd** [comment]
I'll have a look @walidmujahid the library we are using, `django-haystack`, supports elastic search 2, and 5, the live deployment uses the same engine as in the settings.

What do you have as the HAYSTACK_ELASTIC_SEARCH_URL in your env? ? try setting it the to `elasticsearch:9200`

**Comment by walidmujahid** [comment]
> @walidmujahid are there any bugs coming from the web_web container? you should be able to run the web container with out connecting to elasticserach

@owocki I was not fully aware about this option, but I have gone ahead an did this. You are right :-)



> What do you have as the HAYSTACK_ELASTIC_SEARCH_URL in your env? ? try setting it the to `elasticsearch:9200`

@androolloyd Well, I had not set it at all. I went ahead and set it it to `elasticsearch:9200` in the env file, however, it did not work. Furthermore, it seem that the elasticsearch container is causing the containers to hang/wait/not run fully causing the error in localhost:8000. However, by stopping the elasticsearch container, the worker and web containers seem to start running fully, thus, becoming available.

**Comment by walidmujahid** [comment]
> @walidmujahid are there any bugs coming from the web_web container? you should be able to run the web container with out connecting to elasticserach

@owocki I did not fully think that through, but I have gone ahead an did this. You are right :-)

> What do you have as the HAYSTACK_ELASTIC_SEARCH_URL in your env? ? try setting it the to `elasticsearch:9200`

@androolloyd Well, I had not set it at all. I went ahead and set it it to `elasticsearch:9200` in the env file, however, it did not work. Furthermore, it seems that the elasticsearch container is causing the other containers to hang/wait/not run fully causing the error in localhost:8000. However, by stopping the elasticsearch container, the worker and web containers seem to start running fully, thus, becoming available.
